src/Symfony/Component/Validator/Constraints/Ip.php

```diff
@@ -48,7 +48,7 @@ class Ip extends Constraint
 
     public const INVALID_IP_ERROR = 'b1b427ae-9f6f-41b0-aa9b-84511fbb3c5b';
 
-    protected static $versions = [
+    protected const VERSIONS = [
```

Comment:
Isn't this a BC break? A subclass might change the behavior of the validator by modifying the `$versions` variable.

```suggestion
    /** @deprecated */
    protected static $versions = self::VERSIONS;

    protected const VERSIONS = [
```

Reply:
right, updated!